src/components/navbar/index.tsx

```diff
@@ -0,0 +1,152 @@
+import Typography from '@material-ui/core/Typography';
+import InputBase from '@material-ui/core/InputBase';
+import List from '@material-ui/core/List';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemText from '@material-ui/core/ListItemText';
+import MenuItem from '@material-ui/core/MenuItem';
+import Menu from '@material-ui/core/Menu';
+import Button from '@material-ui/core/Button';
+import { ActionCreators as UndoActionCreators } from 'redux-undo';
+
+import { changeImage } from '../../store/modules/cover/actions';
+import images from '../../config/images';
+import historyBehaviorService from '../../services/historyBehaviorService';
+
+import useStyles from './styles';
+import StepsSelector from './selectors';
+
+interface ImageType {
+  id: number,
+  title: string,
+  path: string,
+}
+
+const Navbar: React.FC = (): React.ReactElement => {
+  const classes = useStyles();
+  const dispatch = useDispatch();
+
+  const { pastLength, futureLength } = useSelector(StepsSelector);
+
+  const [isOpenMenu, setIsOpenMenu] = React.useState<null | HTMLElement>(null);
+
+  const clickListHandler = useCallback((event: React.MouseEvent<HTMLElement>): void => {
+    setIsOpenMenu(event.currentTarget);
+  }, []);
+
+  const clickMenuHandler = (image: ImageType) => {
+    if (image) {
+      dispatch(changeImage(image.id));
+    }
+
+    setIsOpenMenu(null);
+  };
+
+  const closeHandler = useCallback((): void => {
+    setIsOpenMenu(null);
+  }, []);
+
+  const pverStepHandler = useCallback((): void => {
+    dispatch(UndoActionCreators.undo());
+  }, []);
+
+  const nextStepHandler = useCallback((): void => {
+    dispatch(UndoActionCreators.redo());
+  }, []);
+
+  const isPrevDisabled = useMemo((): boolean => (
+    pastLength <= 0
+  ), [pastLength]);
+
+  const isFutureDisabled = useMemo((): boolean => (
+    futureLength <= 0
+  ), [futureLength]);
+
+  const historyHandler = useCallback((event: KeyboardEvent): void => {
+    historyBehaviorService.handleKeyPress(event.ctrlKey, event.keyCode);
+  }, []);
+
+  useEffect(() => {
+    window.addEventListener('keydown', historyHandler);
+
+    return () => {
+      window.removeEventListener('keydown', historyHandler);
+    };
+  }, []);
+
+  return (
+    <div className={classes.grow}>
+      <AppBar position="static">
+        <Toolbar>
+          <Typography className={classes.title} variant="h6" noWrap>
+            Material-UI
+          </Typography>
+          <div className={classes.search}>
+            <div className={classes.searchIcon}>
+              <SearchIcon />
+            </div>
+            <InputBase
+              placeholder="Search…"
+              classes={{
+                root: classes.inputRoot,
+                input: classes.inputInput,
+              }}
+              inputProps={{ 'aria-label': 'search' }}
+            />
+          </div>
+          <div>
+            <List component="nav" aria-label="Device settings">
```

Comment:
can we create an additional component to render images dropdown? or you can create a method for this logic